systemd/ocp-mco-sshkey.sh

```diff
@@ -24,17 +24,23 @@ fi
 wait_for_resource_or_die machineconfig/99-master-ssh
 
 echo "Updating the public key resource for machine config operator"
-pub_key=$(cat "$CRC_PUB_KEY_PATH" | tr -d '\n\r')
 
-jq -n --arg pub_key "${pub_key}" '
+# Use --slurpfile to read the key file directly into a jq variable named 'pub_key'.
+# The key's content is never exposed as a command-line argument.
+# We also use jq's rtrimstr function to remove any trailing newlines from the file,
+# which is equivalent to the old 'tr -d' command.
+jq -n --slurpfile pub_key "$CRC_PUB_KEY_PATH" '
 {
   "spec": {
     "config": {
       "passwd": {
         "users": [
           {
             "name": "core",
-            "sshAuthorizedKeys": [ $pub_key ]
+            "sshAuthorizedKeys": [
+              # Trim trailing newlines and carriage returns from the slurped file content
+              $pub_key | rtrimstr("\n") | rtrimstr("\r")
+            ]
```

Comment:
Is it some security venerability of exposing pub key?

Reply:
no, IMO it's not a leak if the pub key is exposed, but I consider it best practice to avoid publishing it for no reason
(may contain a user address, may disclose the encryption strength, may help for a man-in-the-middle attack ....)